### PR TITLE
Enable bundle removal for FBC operators

### PR DESCRIPTION
**Pull request by Allda**
A user can remove a bundle from repository of it uses FCB workflow and the removed bundle is not used in any of existing catalogs. A user can also combine a bundle removal and catalog updates so they can remove entire bundle and catalog at the same time.

JIRA: ISV-5138

### CODE REVIEW

**Review by BorekZnovustvoritel — APPROVED** [review]
I really like using dataclasses instead of dictionaries. I have provided several rather minor comments to double-check but none of them is a blocker from my side. LGTM!

File: operator-pipeline-images/operatorcert/parsed_file.py [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+                )
+
+    def _fetch_bundles_from_catalog_file(
+        self, operator_catalog: OperatorCatalog
+    ) -> list[Dict[str, Any]]:
+        """
+        Fetch bundles from the catalog file
+        """
+        catalog_content_path = operator_catalog.catalog_content_path
+        bundles = []
```
[the remark this answers is not in the document]
Comment (Allda) [reply operator-pipeline-images/operatorcert/parsed_file.py]: Fixed

File: operator-pipeline-images/operatorcert/parsed_file.py [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+        )
+        if len(catalog_operators) > 1:
+            self.errors.append(
+                f"The PR affects more than one catalog operator: {catalog_operators}"
+            )
+
+    def validate(self) -> None:
+        """
+        Run a test suite to validate the changes
+        A test suite is defined as a set of functions that start with "check_"
```
[the remark this answers is not in the document]
Comment (Allda) [reply operator-pipeline-images/operatorcert/parsed_file.py]: Fixed

File: operator-pipeline-images/operatorcert/parsed_file.py [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+    def _fetch_bundles_from_catalog_file(
+        self, operator_catalog: OperatorCatalog
+    ) -> list[Dict[str, Any]]:
+        """
+        Fetch bundles from the catalog file
+        """
+        catalog_content_path = operator_catalog.catalog_content_path
+        bundles = []
+        with open(catalog_content_path, "r", encoding="utf-8") as catalog_file:
+            yaml_content = yaml.safe_load_all(catalog_file)
```
[the remark this answers is not in the document]
Comment (Allda) [reply operator-pipeline-images/operatorcert/parsed_file.py:274]: The operator_repo library already verifies before this line that the file exists and is ok so we don't have to take care of it here.

File: operator-pipeline-images/operatorcert/parsed_file.py [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+            **(self.affected_operators.to_dict()),
+            **(self.affected_bundles.to_dict()),
+            **(self.affected_catalogs.to_dict()),
+            **(self.affected_catalog_operators.to_dict()),
+            "extra_files": list(self.extra_files),
+        }
+        self.enrich_result(result)
+        return result
+
+    def enrich_result(self, result: dict[str, Any]) -> None:
```
[the remark this answers is not in the document]
Comment (Allda) [reply operator-pipeline-images/operatorcert/parsed_file.py]: Fixed

**Review by mantomas — APPROVED** [review]
One nitpick and after addressing Marek's comments LGTM
Is there a plan to cover this new functionality with integration test scenario?

**Comment by Allda** [comment]
> One nitpick and after addressing Marek's comments LGTM Is there a plan to cover this new functionality with integration test scenario?

I wasn't thinking about it because to remove something from a repo you have to add it first by the official way. It makes this test very long because all pipelines need to run twice.

**Comment by mantomas** [comment]
> I wasn't thinking about it because to remove something from a repo you have to add it first by the official way. It makes this test very long because all pipelines need to run twice.
Makes sense - one option might be to extend the current FBC tests, just revert the change afterwards?.. Not sure about the complexity though.

**Pull request merged**